Sources/ServiceLifecycle/ServiceRunnerError.swift

```diff
@@ -0,0 +1,127 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftServiceLifecycle open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftServiceLifecycle project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftServiceLifecycle project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// Errors thrown by the ``ServiceRunner``.
+public struct ServiceRunnerError: Error, Hashable, Sendable {
+    /// A struct representing the possible error codes.
+    public struct Code: Hashable, Sendable, CustomStringConvertible {
+        private enum _Code: Hashable, Sendable {
+            case alreadyRunning
+            case alreadyFinished
+            case serviceFinishedUnexpectedly
+        }
+
+        private var code: _Code
+
+        private init(code: _Code) {
+            self.code = code
+        }
+
+        public var description: String {
+            switch self.code {
+            case .alreadyRunning:
+                return "The service runner is already running the services."
+            case .alreadyFinished:
+                return "The service runner has already finished running the services."
+            case .serviceFinishedUnexpectedly:
+                return "A service has finished unexpectedly."
+            }
+        }
+
+        /// Indicates that the service runner is already running.
+        public static let alreadyRunning = Code(code: .alreadyRunning)
+        /// Indicates that the service runner has already finished running.
+        public static let alreadyFinished = Code(code: .alreadyFinished)
+        /// Indicates that a service finished unexpectedly.
+        public static let serviceFinishedUnexpectedly = Code(code: .serviceFinishedUnexpectedly)
+    }
+
+    /// Internal class that contains the actual error code.
+    private final class Backing: Hashable, Sendable, CustomStringConvertible {
```

Comment:
I don't think `Backing` should implement `CustomStringConvertible`. I think the error itself should implement `CustomStringConvertible`.